token-metadata/program/tests/collection_migrations.rs

```diff
@@ -774,4 +779,360 @@ mod collection_migrations {
+                collection_a_me.pubkey,
+                None,
+            )
+            .await
+            .unwrap();
+
+        // Now we can migrate it over to collection b.
+        nft_item_one
+            .set_and_verify_collection(
+                &mut context,
+                collection_b_nft.pubkey,
+                &kp,
+                kp.pubkey(),
+                collection_b_nft.mint.pubkey(),
+                collection_b_me.pubkey,
+                None,
+            )
+            .await
+            .unwrap();
+
+        let account = get_account(&mut context, &nft_item_one.pubkey).await;
+        let md = TmMetadata::safe_deserialize(&account.data).unwrap();
+
+        let collection = md.collection.unwrap();
+
+        assert_eq!(collection.key, collection_b_nft.mint.pubkey());
+        assert!(collection.verified);
+    }
+
+    #[tokio::test]
+    pub async fn burned_collection_parent_sized_collection() {
+        let mut context = program_test().start_with_context().await;
+
+        // Sized collection
+        let (collection_a_nft, collection_a_me) =
+            Metadata::create_default_sized_parent(&mut context)
+                .await
+                .unwrap();
+
+        // Sized collection
+        let (collection_b_nft, collection_b_me) =
+            Metadata::create_default_sized_parent(&mut context)
+                .await
+                .unwrap();
+
+        let (nft_item_one, _me_item_one) =
+            Metadata::create_default_nft(&mut context).await.unwrap();
+        let (nft_item_two, _me_item_two) =
+            Metadata::create_default_nft(&mut context).await.unwrap();
+
+        let kpbytes = &context.payer;
+        let kp = Keypair::from_bytes(&kpbytes.to_bytes()).unwrap();
+
+        // Add the items to Collection A
+        nft_item_one
+            .set_and_verify_sized_collection_item(
+                &mut context,
+                collection_a_nft.pubkey,
+                &kp,
+                kp.pubkey(),
+                collection_a_nft.mint.pubkey(),
+                collection_a_me.pubkey,
+                None,
+            )
+            .await
+            .unwrap();
+
+        nft_item_two
+            .set_and_verify_sized_collection_item(
+                &mut context,
+                collection_a_nft.pubkey,
+                &kp,
+                kp.pubkey(),
+                collection_a_nft.mint.pubkey(),
+                collection_a_me.pubkey,
+                None,
+            )
+            .await
+            .unwrap();
+
+        // Burn the collection A parent.
+        burn(
+            &mut context,
+            collection_a_nft.pubkey,
+            &kp,
+            collection_a_nft.mint.pubkey(),
+            collection_a_nft.token.pubkey(),
+            collection_a_me.pubkey,
+            None,
+        )
+        .await
+        .unwrap();
+
+        // This should succeed.
+        nft_item_one
+            .unverify_collection(
```

Comment:
It may be slightly confusing that this handler works with a burned sized parent, because yeah I know the collection is gone, but it seems like if I was trying to use this even if I knew it was burned, I might reach for `unverify_sized_collection` first.  It might be worth considering making that handler able to handle burned parents as well, what do you think?

Reply:
wdyt about this comment above?

Reply:
added in the latest commit